cmd/thanos/store.go

```diff
@@ -379,6 +382,25 @@ func runStore(
 			cancel()
 		})
 	}
+
+	infoSrv := info.NewInfoServer(
```

Comment:
What's the deprecation plan for ths old Info methods? Can we unify implementation somehow at least?

Reply:
ping

Reply:
Really sorry for delay.
Here is the deprecation plan I had in mind.
1. Remove old flags (i.e. `--store`, `--rule`, `--metadata`, `--exemplar`, `--target`) and all the boilerplate code we have for all commands. ( eg. duplication check, dnsProviders ...)
2. Remove `storeset.go` file ( I am working on to move new endpoint flow in new file as discussed [here](https://github.com/thanos-io/thanos/pull/4282#discussion_r653353874)  ) and all dependencies of this file from the code base.
3.  Remove [Info](https://github.com/thanos-io/thanos/blob/bc8421ac181579f1a8b2bf99183667fba965381e/pkg/store/storepb/rpc.proto#L30) rpc from proto file of Store API and all its corresponding implementations. ( to be specific there 5 different implementations we have of [StoreClient](https://github.com/thanos-io/thanos/blob/bc8421ac181579f1a8b2bf99183667fba965381e/pkg/store/storepb/rpc.pb.go#L692) - `BucketStore`, `MultiTSDBStore`, `PrometheusStore`, `ProxyStore` and `TSDBStore`) 

All these steps will be done in single PR. I think this plan should work . 
Please let me know if im missing something :)

Reply:
>  Can we unify implementation somehow at least?

Why do we need to unify all info methods?

Reply:
1. When we want to change flag we usually deprecate flag first (add in help that it will be removed after next 2 releases). Then add new flags that works simultaneously. Fail if both new and old way is used. Map old flags to new one really before starting the program.
2. Agree
3. Generally makes sense, but we need to make sure that e.g old Querier having a client against old StoreAPI will work with new InfoAPI. Probably we will need to leave old Info Method (add deprecated comment) in old APIs. In the same time the new Querier should work with old APIs.
   * Potential flow: 1. Try (reflect)? InfoAPI end endpoint. If gRPC server does not support it we fallback to store, if store does it work then ruler ..... only for first call / error.
   * e2e test
   * IMPORTANT: This does not mean we need to implement separate info method for every server. We can implement one Info method and use the same one for both InfoAPI and e.g StoreAPI.